ncm-sysctl/src/main/perl/sysctl.pm

```diff
@@ -22,85 +22,109 @@ use NCM::Component;
 our @ISA = qw(NCM::Component);
 use LC::Exception qw(throw_error);
 our $EC=LC::Exception::Context->new->will_store_all;
+our $NoActionSupported = 1;
 
+use CAF::FileWriter;
+use CAF::Process;
 use EDG::WP4::CCM::Element;
-
 use NCM::Check;
-use LC::Process qw(trun);
-
-# For convenience
-my $base = '/software/components/sysctl';
-
 
-##########################################################################
-sub Configure {
-##########################################################################
+sub Configure
+{
     my ($self,$config)=@_;
-    
+
     # Load config into a hash
-    my $sysctl_config = $config->getElement($base)->getTree();
+    my $sysctl_config = $config->getElement($self->prefix)->getTree();
     my $variables = $sysctl_config->{variables};
     my $configFile = $sysctl_config->{confFile};
     my $changes;
-    unless ( $configFile ) {
-	$self->error('Sysctl configuration file not defined');
-    }
-
-    unless ($configFile =~ m{^(/.+)$}) {
-	throw_error("Invalid configuration file in the profile: $configFile");
-	return();
-    }
-    $configFile = $1;
     my $sysctl_exe = $sysctl_config->{command};
-    unless ( $sysctl_exe ) {
-	$self->error('Sysctl command not defined');
-    }
-
     unless ($sysctl_exe =~ m{^(/\S+)$}) {
-	throw_error("Invalid sysctl command on the profile: $sysctl_exe");
-	return();
+        throw_error("Invalid sysctl command on the profile: $sysctl_exe");
+        return();
     }
 
     $sysctl_exe = $1;
 
     unless (-x $sysctl_exe) {
-	$self->error ("$sysctl_exe not found");
-	return;
-    }
-
-    unless (-e $configFile && -w $configFile) {
-	$self->warn("Sysctl configuration file does not exist ",
-		    "or is not writable ($configFile)");
-	return;
+        $self->error ("$sysctl_exe not found");
+        return;
     }
 
-    foreach my $key (sort(keys(%$variables))) {
-	my $value = $variables->{$key};
-	my $st = NCM::Check::lines($configFile,
-				   backup => '.old',
-				   linere => '#?\s*'.$key.'\s*=.*',
-				   goodre => '\s*'.$key.'\s*=\s*'.$value,
-				   good => "$key = $value",
-				   add => 'last'
-				  );
-	if ($st < 0) {
-	    $self->error("Failed to update sysctl $key (value=$value)");
-	} else {
-	    $changes += $st;
-	}
+    if ($configFile =~ m{/}) {
+        $self->debug(1, "confFile setting is a path");
+        $changes = $self->sysctl_file($configFile, $sysctl_exe, $variables);
+    } else {
+        $self->debug(1, "confFile is a relative filename, using sysctl.d");
+        $changes = $self->sysctl_dir($configFile, $sysctl_exe, $variables);
+        # update the configFile path for the sysctl execution
+        $configFile = "/etc/sysctl.d/$configFile";
```

Comment:
@ned21 remark wrt the file name. i think it would be better to make this 
```perl
$configFile = "/etc/sysctl.d/zzz-quattor-$configFile";
```
so on reboot, the quattor changes are always winning.